plugins/workspace-plugin/src/ca-cert.ts

```diff
@@ -0,0 +1,53 @@
+/**********************************************************************
+ * Copyright (c) 2019-2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+const PUBLIC_CRT_PATH = '/public-certs';
+const SS_CRT_PATH = '/tmp/che/secret/ca.crt';
+
+const CA_BUNDLE_PATH = '/tmp/ca-bundle.crt';
+
+let bundle: string | undefined = undefined;
+
+export async function getAuthorityCertificate(): Promise<string | undefined> {
+  if (bundle !== undefined) {
+    return bundle;
+  }
+
+  const certificateAuthority: Buffer[] = [];
+
+  if (fs.existsSync(SS_CRT_PATH)) {
```

Comment:
await fs.exists(

Reply:
It's deprecated
```
    /**
     * Asynchronously tests whether or not the given path exists by checking with the file system.
     * @deprecated
     * @param path A path to a file or directory. If a URL is provided, it must use the `file:` protocol.
     * URL support is _experimental_.
     */
    function exists(path: PathLike, callback: (exists: boolean) => void): void;
```

Reply:
https://github.com/jprichardson/node-fs-extra/blob/master/docs/pathExists.md

Reply:
Reworked

Reply:
The root was: use fs-extra instead fs :)